contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -82,182 +138,256 @@ contract MixinStakingPoolRewards is
-        //   2. The current epoch is zero: delegation begins at epoch 1
-        if (unsyncedDelegatedStakeToPoolByOwner.currentEpoch == currentEpoch || currentEpoch == 0) return 0;
-
-        // compute reward accumulated during `delegatedStake.currentEpoch`;
-        uint256 rewardsAccumulatedDuringLastStoredEpoch = (unsyncedDelegatedStakeToPoolByOwner.currentEpochBalance != 0)
-            ? _computeMemberRewardOverInterval(
-                poolId,
-                unsyncedDelegatedStakeToPoolByOwner.currentEpochBalance,
-                uint256(unsyncedDelegatedStakeToPoolByOwner.currentEpoch).safeSub(1),
-                unsyncedDelegatedStakeToPoolByOwner.currentEpoch
-            )
-            : 0;
+        // There can be no rewards in epoch 0 because there is no delegated
+        // stake.
+        if (currentEpoch == 0) {
+            return 0;
+        }
 
-        // compute the reward accumulated by the `next` balance;
-        // this starts at `delegatedStake.currentEpoch + 1` and goes up until the last epoch, during which
-        // rewards were accumulated. This is at most the most recently finalized epoch (current epoch - 1).
-        uint256 rewardsAccumulatedAfterLastStoredEpoch = (_cumulativeRewardsByPoolLastStored[poolId] > unsyncedDelegatedStakeToPoolByOwner.currentEpoch)
-            ? _computeMemberRewardOverInterval(
+        // There can be no rewards if the last epoch when stake was synced is
+        // equal to the current epoch, because all prior rewards, including
+        // rewards finalized this epoch have been claimed.
+        if (unsyncedStake.currentEpoch == currentEpoch) {
+            return 0;
+        }
+
+        // If there are unfinalized rewards this epoch, compute the member's
+        // share.
+        if (unfinalizedMembersReward != 0 && unfinalizedMembersStake != 0) {
+            // Unfinalized rewards are always earned from stake in
+            // the prior epoch so we want the stake at `currentEpoch-1`.
+            uint256 _stake = unsyncedStake.currentEpoch >= currentEpoch.safeSub(1) ?
+                unsyncedStake.currentEpochBalance :
+                unsyncedStake.nextEpochBalance;
+            if (_stake != 0) {
+                reward = LibMath.getPartialAmountFloor(
+                    unfinalizedMembersReward,
+                    unfinalizedMembersStake,
+                    _stake
+                );
+            }
+        }
+
+        // Get the last epoch where a reward was credited to this pool, which
+        // also happens to be when we last created a cumulative reward entry.
+        uint256 lastRewardEpoch = _cumulativeRewardsByPoolLastStored[poolId];
+
+        // If the stake has been touched since the last reward epoch,
+        // it has already been claimed.
+        if (unsyncedStake.currentEpoch >= lastRewardEpoch) {
+            return reward;
+        }
+        // From here we know: `unsyncedStake.currentEpoch < currentEpoch > 0`.
```

Comment:
It looks like the intervals have been shifted. Is this on purpose?
```
# Previously the ranges were:
[unsyncedStake.currentEpoch-1 .. unsyncedStake.currentEpoch]
[unsyncedStake.currentEpoch .. lastRewardEpoch]

# Now the ranges are:
[unsyncedStake.currentEpoch .. unsyncedStake.currentEpoch + 1]
[unsyncedStake.currentEpoch + 1 .. lastRewardEpoch + 1]
```

Reply:
Yup. This is from crediting rewards at the top of an epoch rather than at the end, and folding in unfinalized rewards.